meteor/client/ui/Settings/ShowStyle/VariantSettings.tsx

```diff
@@ -35,12 +42,102 @@ export const ShowStyleVariantsSettings = withTranslation()(
 
 			this.state = {
 				editedMappings: [],
+				uploadFileKey: Date.now(),
+			}
+		}
+
+		private importShowStyleVariants = (event: React.ChangeEvent<HTMLInputElement>): void => {
+			const { t } = this.props
+
+			const file = event.target.files ? event.target.files[0] : null
+			if (!file) {
+				return
 			}
+
+			const reader = new FileReader()
+			reader.onload = (progressEvent: ProgressEvent<FileReader>) => {
+				this.setState({
+					uploadFileKey: Date.now(),
+				})
+
+				const uploadFileContents = (progressEvent.target as any).result
+
+				const newShowStyleVariants: ShowStyleVariant[] = []
+				try {
+					JSON.parse(uploadFileContents).map((showStyleVariant) => newShowStyleVariants.push(showStyleVariant))
+					if (!_.isArray(newShowStyleVariants)) {
+						throw new Error('Imported file did not contain an array')
+					}
+				} catch (error) {
+					NotificationCenter.push(
+						new Notification(
+							undefined,
+							NoticeLevel.WARNING,
+							t('Failed to import new showstyle variants: {{errorMessage}}', { errorMessage: error + '' }),
+							'VariantSettings'
+						)
+					)
+					return
+				}
+
+				this.importShowStyleVariantsFromArray(newShowStyleVariants)
+			}
+			reader.readAsText(file)
+		}
+
+		private importShowStyleVariantsFromArray = (showStyleVariants: ShowStyleVariant[]): void => {
+			const { t } = this.props
+			showStyleVariants.forEach((showStyleVariant: ShowStyleVariant) => {
+				MeteorCall.showstyles.insertShowStyleVariantWithProperties(showStyleVariant, showStyleVariant._id).catch(() => {
+					NotificationCenter.push(
+						new Notification(
+							undefined,
+							NoticeLevel.WARNING,
+							t('Failed to import Variant {{name}}. Make sure it is not already imported.', {
+								name: showStyleVariant.name,
+							}),
+							'VariantSettings'
+						)
+					)
+				})
+			})
 		}
-		isItemEdited = (layerId: ProtectedString<any>) => {
+
+		private copyShowStyleVariant = (showStyleVariant: ShowStyleVariant): void => {
+			showStyleVariant.name = 'Copy of ' + showStyleVariant.name
```

Comment:
Just a suggestion to use string interpolation
```suggestion
			showStyleVariant.name = `Copy of ${showStyleVariant.name}`
```

Reply:
Added it to [f74f966](https://github.com/tv2/tv-automation-server-core/pull/188/commits/f74f9665479993af5c8647b78f40a11fe9370e0f).